### ISSUE TITLE
Not OK: perl v5.8.0 +DEVEL17462 on i386-freebsd 4.6-release    (UNINSTALLED)

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5710]
Migrated from [rt.perl.org#10038](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=10038) (status was 'resolved')

Searchable as RT10038$

**Comment by p5pRT** [comment]
### From @nwc10



$ \./perl lib/Net/t/hostname\.t 
1\.\.2
not ok 1
Use of uninitialized value in string eq at lib/Net/t/hostname\.t line 39\.
Use of uninitialized value in string eq at lib/Net/t/hostname\.t line 39\.
ok 2

The uninitialized value warning is from this line&#8203;:

&nbsp;  my @&#8203;dummy = grep \{ hostname eq $\_ \} @&#8203;domain;

The failure of test 1 is before this\, and is because the domainname isn't set
on this machine \(a side effect of how I upgraded to FreeBSD 4\.6\)

If I run the domainname command it returns nothing\. Its man page says&#8203;:

DESCRIPTION
&nbsp;     The domainname utility prints the name of the current YP/NIS domain\.  The
&nbsp;     super\-user can set the domain name by supplying an argument; this is usu­
&nbsp;     ally done in the network initialization script /etc/rc\.network\, normally
&nbsp;     run at boot time\.

so I believe it is legal for a machine not to have a domain\, if it's not using
NIS\. Therefore I don't believe that this test should fail if there is no
domain name\. Quite how one writes a test that distinguishes between there
being no domainname\, and there being one and the module failing to return it\,
I don't know\.

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=none

Site configuration information for perl v5.8.0:

Configured by nick at Wed Jul 10 19:39:54 GMT 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0 patch 17465) configuration:
  Platform:
    osname=freebsd, osvers=4.6-release, archname=i386-freebsd
    uname='freebsd thinking-cap.moo 4.6-release freebsd 4.6-release #0: tue jun 11 06:14:12 gmt 2002 murray@builder.freebsdmall.com:usrsrcsyscompilegeneric i386 '
    config_args='-Dcc=ccache gcc -Dld=gcc -Dusedevel -Ubincompat5005 -Uinstallusrbinperl -Dcf_email=nick@ccl4.org -Dperladmin=nick@ccl4.org -Dinc_version_list=  -Dinc_version_list_init=0 -Doptimize=-g -Dinstallman1dir=none -Dinstallman3dir=none -de -Dprefix=/stuff'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache gcc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DDEBUGGING -fno-strict-aliasing -I/usr/local/include',
    optimize='-g',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -DDEBUGGING -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='2.95.3 20010315 (release) [FreeBSD]', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-Wl,-E  -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lm -lc -lcrypt -lutil
    perllibs=-lm -lc -lcrypt -lutil
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fpic', lddlflags='-shared  -L/usr/local/lib'

Locally applied patches:
    DEVEL17462


@INC for perl v5.8.0:
    lib
    /stuff/lib/perl5/5.8.0/i386-freebsd
    /stuff/lib/perl5/5.8.0
    /stuff/lib/perl5/site_perl/5.8.0/i386-freebsd
    /stuff/lib/perl5/site_perl/5.8.0
    /stuff/lib/perl5/site_perl
    .


Environment for perl v5.8.0:
    HOME=/home/nick
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=en_GB.ISO_8859-1
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/nick/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/usr/X11R6/bin:/home/nick/bin:/sbin:/usr/sbin:/usr/local/sbin
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/bash


```
</details>

**Comment by p5pRT** [comment]
### From @jhi

This got fixed by change \#17765\, marking the ticket as resolved\.

**Comment by p5pRT** [comment]
@jhi - Status changed from 'open' to 'resolved'
